dev/devicelab/lib/framework/metrics_center.dart

```diff
@@ -0,0 +1,115 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:metrics_center/metrics_center.dart';
+import 'package:path/path.dart' as p;
+
+Future<ProcessResult> runGit(
+  List<String> args, {
+  String? processWorkingDir,
+}) async {
+  return Process.run(
+    'git',
+    args,
+    workingDirectory: processWorkingDir,
+    runInShell: true,
+  );
+}
+
+/// Returns commit time in committer date in UNIX timestamp.
+Future<String> getCommitDate() async {
+  final String gitRoot = p.absolute('.');
+  final ProcessResult logResult = await runGit(
+    <String>['log', '--pretty=format:%ct', '-n', '1'],
+    processWorkingDir: gitRoot,
+  );
+  if (logResult.exitCode != 0) {
+    throw 'Unexpected exit code ${logResult.exitCode}';
+  }
+  return logResult.stdout.toString().trim();
+}
+
+/// Authenticate and connect to gcloud storage.
+///
+/// It supports both token and credential authentications.
+Future<FlutterDestination> connectFlutterDestination() async {
+  const String kTokenPath = 'TOKEN_PATH';
+  const String kGcpProject = 'GCP_PROJECT';
+  final Map<String, String> env = Platform.environment;
+  final bool isTesting = env['IS_TESTING'] == 'true';
+  if (env.containsKey(kTokenPath) && env.containsKey(kGcpProject)) {
+    return FlutterDestination.makeFromAccessToken(
+      File(env[kTokenPath]!).readAsStringSync(),
+      env[kGcpProject]!,
+      isTesting: isTesting,
+    );
+  }
+  return FlutterDestination.makeFromCredentialsJson(
+    jsonDecode(env['BENCHMARK_GCP_CREDENTIALS']!) as Map<String, dynamic>,
```

Comment:
Are able to reuse the LUCI account token?

Reply:
@yusufm did we manage to implement the LUCI context manager in flutter/flutter devicelab classes?

Reply:
Yes. For prod runs, we add env variable of `TOKEN_PATH `and `GCP_PROJECT` from recipe, and use the LUCI account token. (a CL in process).